From 48292a413d991a8407da0f984dfa6919389ae6de Mon Sep 17 00:00:00 2001
From: Matthew Shyu <matthew.shyu@amlogic.com>
Date: Mon, 1 Jun 2015 12:01:11 +0800
Subject: [PATCH 5625/5965] Crypto: Zero out HW AES pio fields.

Secure boot will put HW AES mode into pio modes and thus causing dma
mode fail

Change-Id: I7bc93c98f95a7e750db6d784c445a2edf8464ca8
---
 drivers/amlogic/crypto/aml-aes.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/drivers/amlogic/crypto/aml-aes.c b/drivers/amlogic/crypto/aml-aes.c
index 73840d1c047d..f429a1b2cafa 100644
--- a/drivers/amlogic/crypto/aml-aes.c
+++ b/drivers/amlogic/crypto/aml-aes.c
@@ -144,9 +144,7 @@ static void set_aes_key_normal(u32 *key, unsigned int keylen)
 static void set_aes_key_secure(u32 *key, unsigned int keylen)
 {
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_REG0,
-			(secure_apb_slot3_rd_reg(SEC_BLKMV_AES_REG0)
-			& ~(0x3 << 4))
-			| (AES_THREAD_INDEX << 4));
+			(AES_THREAD_INDEX << 4));
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_KEY_0, *key);
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_KEY_1, *(key+1));
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_KEY_2, *(key+2));
@@ -194,9 +192,7 @@ static void set_aes_iv_big_endian_normal(unsigned long *iv)
 static void set_aes_iv_secure(unsigned long *iv)
 {
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_REG0,
-			(secure_apb_slot3_rd_reg(SEC_BLKMV_AES_REG0)
-			& ~(0x3 << 4))
-			| (AES_THREAD_INDEX << 4));
+			(AES_THREAD_INDEX << 4));
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_IV_0, *iv);
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_IV_1, *(iv+1));
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_IV_2, *(iv+2));
@@ -206,9 +202,7 @@ static void set_aes_iv_secure(unsigned long *iv)
 static void set_aes_iv_big_endian_secure(unsigned long *iv)
 {
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_REG0,
-			(secure_apb_slot3_rd_reg(SEC_BLKMV_AES_REG0)
-			& ~(0x3 << 4))
-			| (AES_THREAD_INDEX << 4));
+			(AES_THREAD_INDEX << 4));
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_IV_3, swap_ulong32(*iv));
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_IV_2, swap_ulong32(*(iv+1)));
 	secure_apb_slot3_wr_reg(SEC_BLKMV_AES_IV_1, swap_ulong32(*(iv+2)));
-- 
2.19.0

